### ISSUE TITLE
Chipmunk/Javascript bindings - can't use getBody

### ISSUE THREAD

**Issue by Tokyonono** [cocos2d/cocos2d-x#2274]
Hi everyone,

I got a problem with Chipmunk when using the Javascript code I wrote for Cocos2d-html5 (which works fine on Chrome/Firefox) on iPhone.
It seems I can't call getBody() on cc.PhysicsSprite, so I can't apply any impulse or force.
How can I apply forces/impulses to bodies if I can't get the body?

Furthermore, I needed to overwrite cp.Body.velocity_func (I followed the recommendation of the author of Chipmunk himself) to achieve what I wanted to do (basically, put a different velocity limit on X an Y axis).
Is there any way I can get that to work too?

Any help, current status of the implementation or work around would be welcome.
Thanks!

**Comment by Tokyonono** [comment]
Any solution, workaround or fix for this?

**Comment by Tokyonono** [comment]
Nobody reads bug reports on Github, right? (at least not in the past 4 months) ... That is sad. I hope some day you will realise that it is important to interact with the community. Great framework anyway.

**Comment by dumganhar** [comment]
I am fixing it. Please wait and then help to test it.

**Comment by dumganhar** [comment]
Fixed at https://github.com/cocos2d/cocos2d-x/pull/3163 and https://github.com/cocos2d/cocos2d-x/pull/2869
Could you test it now?

**Comment by Tokyonono** [comment]
OK, I will test now and give you feedback ASAP.

**Comment by minggo** [comment]
Thanks @Tokyonono.
Please close it if it is fixed.

**Comment by Tokyonono** [comment]
I cannot test with the current develop version. I cannot compile (even using C++11 compiler and libc++). I got error messages like "CCDeprecated.h is missing." and some others...

**Comment by minggo** [comment]
@Tokyonono 
Did you synchronize latest code?

**Comment by Tokyonono** [comment]
I took the last code on github just now, using git (https://github.com/cocos2d/cocos2d-x/commit/74002d48842cfa58d3e67c226cf398b97a7196d7).
I installed the templates, created a new project (_js), changed to C+11 and libc++.
When I compile : "CCDeprecated.h" file not found.

**Comment by Tokyonono** [comment]
Sorry guys, I tried everything to compile, I updated to the latest version again, it simply won't compile.
Steps (on MAC OS X 10.8.4 [Mountain Lion], Xcode 4.6.3, Target: iOS iPhone 6.1):
1/ Pull the latest version (6d958af23aa444afcf780e4c346da688146686a9)
2/ ./install-templates-xcode.sh -u -f
3/ Create a project (I used cocos2dx_js)
4/ Build => it doesnt't work because std::function is not recognized (I would have created a separate branch of 3.0 preview... but it is just me)
5/ Edit build settings => use C++11 and libc++
6/ Build again => "CCDeprecated.h" file not found
7/ Comment "CCDeprecated.h" file include or copy/paste the file manually
8/ "etc/etc1.h" file not found
9/ I tried some more steps, but never could get it to compile. 2.1.4 works fine though.
